functions/src/definitions/eventHandlers/updateTypesense.ts

```diff
@@ -5,16 +5,19 @@ import {
   CollectionTypes,
 } from "../common/typesense/collectionOperations"
 import { firestoreTimestampToYYYYMM } from "../common/utils"
+import { onDocumentWritten } from "firebase-functions/v2/firestore"
 
-const onMessageWrite = functions
-  .region("asia-southeast1")
-  .runWith({ secrets: ["TYPESENSE_TOKEN"] })
-  .firestore.document("/messages/{messageId}")
-  .onWrite(async (snap, context) => {
+const onMessageWriteV2 = onDocumentWritten(
+  {
+    document: "messages/{messageId}",
+    secrets: ["TYPESENSE_TOKEN"],
+  },
+  async (event) => {
     // Grab the current value of what was written to Firestore.
-    const id = context.params.messageId
+    const id = event.params.messageId
     // If the message is deleted, delete from the Typesense index
-    if (!snap.after.exists) {
+    const messageSnap = event?.data?.after
```

Comment:
```suggestion
    const postMessageSnap = event?.data?.after
```

Similar to `postChangeData`

Reply:
ok addressed this